### PR TITLE
edu-1036 | Python dev guide | Project setup section 

### PR DESCRIPTION
**Pull request by rachfop**
This PR add the Project setup section to the Java SDK dev guide.

Preview: https://temporal-documentation-git-bc-py.preview.thundergun.io/

### CODE REVIEW

File: docs-src/python/backgroundcheck-boilerplate-workflow-details.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+Use the `@workflow.defn` decorator on the `BackgroundCheck` class to identify a Workflow.
+
+Use the `@workflow.run` to mark the entry point method to be invoked. This must be set on one asynchronous method defined on the same class as `@workflow.defn`.
+
+Run methods have positional parameters.
+
+By default Workflows are run in a sandbox to help avoid non-deterministic code.
+If a call that is known to be non-deterministic is performed, an exception will be thrown in the workflow which will "fail the task" which means the workflow will not progress until fixed.
+Import your Activity code into the Workflow Definition using the `with [workflow.unsafe.imports_passed_through()](https://python.temporal.io/temporalio.workflow.unsafe.html#imports_passed_through)` context manager.
```
Comment (flossypurse) [inline docs-src/python/backgroundcheck-boilerplate-workflow-details.md:23]: I think this is out of scope for this section.
I know it it about clarifying Activity imports - but we should avoid the deeper dive into determinism here if we can.

Cab we shorten this to focus on happy path to importing?

File: docs-src/python/choose-dev-cluster.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+If you are eager to to set up Cluster-level metrics for performance tuning, we recommend using a self-hosted Cluster or Temporal Cloud.
+
+#### Start the dev server
+
+**How to start a local development server**
+
+If you have successfully installed the Temporal CLI, open a new terminal and run the following command:
+
+```bash
+temporal server start-dev
```
Comment (taonic) [inline docs-src/python/choose-dev-cluster.md:49]: Would we consider mentioning the support of persistent DB via https://docs.temporal.io/cli/cmd-options#db-filename as an option?

Users in the community do ask questions about how to keep the workflows around in the dev env.

File: docs-src/python/backgroundcheck-boilerplate-self-hosted-worker.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+from activities.ssntraceactivity_dacx import ssn_trace_activity
+from workflows.backgroundcheck_dacx import BackgroundCheck
+
+
+
+async def main():
+    client = await Client.connect(
+        "172.18.0.4:7233"
+    )  # The IP address of the Temporal Server on your network.
```
Comment (cretz) [inline docs-src/python/backgroundcheck-boilerplate-self-hosted-worker.md:40]: This comment is placed after the client and not on the IP field its referring to

File: docs-src/python/backgroundcheck-boilerplate-self-hosted-worker.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import os
+
+from temporalio.client import Client, TLSConfig
+from temporalio.worker import Worker
+
+from activities.ssntraceactivity_dacx import ssn_trace_activity
+from workflows.backgroundcheck_dacx import BackgroundCheck
+
+
+
```
Comment (cretz) [inline docs-src/python/backgroundcheck-boilerplate-self-hosted-worker.md:36]: Inconsistent on whether you have `# ...` between imports and `main` vs some number of blank lines

File: docs-src/python/backgroundcheck-boilerplate-run-a-dev-server-worker.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+:::copycode Sample application code
+
+The following code sample comes from a working and tested sample application.
+The code sample might be abridged within the guide to highlight key aspects.
+Visit the source repository to [view the source code](https://github.com/temporalio/documentation-samples-python/blob/bgc/backgroundcheck_boilerplate/dev_server_worker/main_dacx.py) in the context of the rest of the application code.
+
+:::
+
+```python
+import asyncio
```
Comment (cretz) [inline docs-src/python/backgroundcheck-boilerplate-run-a-dev-server-worker.md:33]: Unused import

File: docs-src/python/backgroundcheck-boilerplate-self-hosted-worker.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+:::copycode Sample application code
+
+The following code sample comes from a working and tested sample application.
+The code sample might be abridged within the guide to highlight key aspects.
+Visit the source repository to [view the source code](https://github.com/temporalio/documentation-samples-python/blob/bgc/backgroundcheck_boilerplate/self_hosted_worker/main_dacx.py) in the context of the rest of the application code.
+
+:::
+
+```python
+import asyncio
```
Comment (cretz) [inline docs-src/python/backgroundcheck-boilerplate-self-hosted-worker.md:26]: Unused import

File: docs-src/python/backgroundcheck-boilerplate-self-hosted-worker.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+import os
```
Comment (cretz) [inline docs-src/python/backgroundcheck-boilerplate-self-hosted-worker.md:27]: Unused import

File: docs-src/python/backgroundcheck-boilerplate-self-hosted-worker.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+from temporalio.client import Client, TLSConfig
```
Comment (cretz) [inline docs-src/python/backgroundcheck-boilerplate-self-hosted-worker.md:29]: `TLSConfig` is an unused import

File: docs-src/python/backgroundcheck-boilerplate-ssntrace-activity.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+- go sdk
+- code sample
+- activity
+---
+
+<!-- DO NOT EDIT THIS FILE DIRECTLY.
+THIS FILE IS GENERATED from https://github.com/temporalio/documentation-samples-python/blob/bgc/backgroundcheck_boilerplate/activities/ssntraceactivity_dacx.py. -->
+
+In the Temporal Python SDK programming model, an Activity is an function.
+You can use asynchronous, synchronous multithreaded, and synchronous multiprocess/other.
```
Comment (cretz) [inline docs-src/python/backgroundcheck-boilerplate-ssntrace-activity.md:16]:
```suggestion
You can use asynchronous, synchronous multithreaded, and synchronous multiprocess/other functions to define an Activity.
```

File: docs-src/python/backgroundcheck-boilerplate-workflow-details.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+To spawn an [Activity Execution](https://python.temporal.io/temporalio.workflow.html#execute_activity), call `execute_activity()` inside your Workflow Definition.
+This API is available from the `workflow` module from the `temporalio` package.
+The `execute_activity` API call requires either `schedule_to_close_timeout` or `start_to_close_timeout`.
+
+In this example, pass in the Activity name, `ssn_trace_activity` and an argument, `ssn`.
```
Comment (cretz) [inline docs-src/python/backgroundcheck-boilerplate-workflow-details.md:29]: I do not see an example in this file (trying to understand the value of breaking up the files)

File: docs-src/python/choose-dev-cluster.md [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+
+However, for most developers we recommend starting by choosing one of the following:
+
+- [Local development server](#local-dev-server)
+- [Temporal Cloud](#temporal-cloud)
+- [Self-hosted Temporal Cluster](#self-hosted-temporal-cluster)
+
+:::info Temporal does not directly run your code
+
+Keep in mind that in every scenario, the “Temporal Platform” does not host and run your Workers (application code).
```
Comment (cretz) [inline docs-src/python/choose-dev-cluster.md:25]: In some docs you use `“` style quotes, and in some `"`

File: docs-src/python/choose-dev-cluster.md [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+It is up to you, the developer, to host your application code.
+The Temporal Platform ensures that properly written code durably executes in the face of platform-level failures.
+
+:::
+
+### Local dev server
+
+**When to use a local development server?**
+
+We recommend using the local development server if you are new to Temporal, or want to start something from scratch and don’t have a self-hosted environment ready or want to pay for a Temporal Cloud account.
```
Comment (cretz) [inline docs-src/python/choose-dev-cluster.md:35]: In some docs you use `’` style apostrophes, and in some `'`

File: docs-src/python/choose-dev-cluster.md [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+temporal server start-dev
+```
+
+This command automatically starts the Temporal Web UI, creates a default Namespace, and creates an in-memory database.
+
+The Temporal Web UI serves to [http://localhost:8233](http://localhost:8233/).
+
+For more command details and options, see the [CLI reference](/cli/server/start-dev)
+
+#### Create a custom Namespace
```
Comment (cretz) [inline docs-src/python/choose-dev-cluster.md:58]: Not sure it's a common use case to create a custom namespace on the local dev server. We offer `--namespace` to change the created default. It's much more common to create a namespace on, say, the self-hosted server.

File: docs-src/python/install-cli.md [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+**How to install the Temporal CLI on Windows**
+
+Follow these instructions to install the Temporal CLI on Windows:
+
+**Install the Temporal CLI from CDN**
+
+1. Select the platform and architecture needed and download the binary.
+
+- Download for Windows amd64: https://temporal.download/cli/archive/latest?platform=windows&arch=amd64
+- Download for Windows arm64: https://temporal.download/cli/archive/latest?platform=windows&arch=arm64
```
Comment (cretz) [inline docs-src/python/install-cli.md:78]: Windows ARM is so rare, it's not worth discussing

File: docs-src/python/project-structure.md [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+However, there are some general ways to think about organizing code.
+
+The best practice is to group Workflows together, Activities together, and separate your Worker process into a standalone file.
+Often this happens respectively per use case, business process, or domain.
+
+For monorepo-style organizational techniques, consider a designated Workflow directory for each use case and place each Workflow in its own file, but also maintain a dedicated place for shared Activities.
+
+For example, your project structure could look like this:
+
+```text
```
Comment (cretz) [inline docs-src/python/project-structure.md:32]: Many of these dirs would have a `__int__.py` but of course don't have to

File: docs-src/python/project-structure.md [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+        /worker
+            | main.py
+    /tests
+       | pytest.ini
+       | __init__.py
+	   | workflow_tests.py
+       | activity_tests.py
+```
+
+If you are following along with this guide, your project will look like this:
```
Comment (cretz) [inline docs-src/python/project-structure.md:58]: You are inconsistent with yourself here. The above listing shows `tests` adjacent with `background_check` and this sample shows `tests` inside `backgroundcheck`. Even if the `tests` remain a bit inconsistent which is fine, at the least the underscores in the name should be consistent

File: docs-src/python/project-structure.md [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+        /workflows
+            | loan_application_workflow.py
+        /activities
+            | credit_check_activity.py
+        /worker
+            | main.py
+    /tests
+       | pytest.ini
+       | __init__.py
+	   | workflow_tests.py
```
Comment (cretz) [inline docs-src/python/project-structure.md:54]: This is showing a file within a file

File: docs-src/python/project-structure.md [hunk 18]
```diff
@@ -0,0 +1,88 @@
...
+            | main.py
+    /loan_application
+        /workflows
+            | loan_application_workflow.py
+        /activities
+            | credit_check_activity.py
+        /worker
+            | main.py
+    /tests
+       | pytest.ini
```
Comment (cretz) [inline docs-src/python/project-structure.md:52]: Lots of people don't use `pytest.ini` anymore, they have `pyproject.toml` at the root

File: docs-src/python/project-structure.md [hunk 19]
```diff
@@ -0,0 +1,88 @@
...
+    /worker
+        | main.py
+    /tests
+       | pytest.ini
+       | __init__.py
+	   | workflow_tests.py
+       | activity_tests.py
+```
+
+### Initialize Python project dependency framework
```
Comment (cretz) [inline docs-src/python/project-structure.md:75]: IMO you should also link to the general Python docs (e.g. https://packaging.python.org/en/latest/tutorials/installing-packages/#creating-virtual-environments) when summarizing general Python documentation

File: docs-src/python/self-hosted-worker-dockerfile.md
Comment (cretz) [inline docs-src/python/self-hosted-worker-dockerfile.md:1]: I think it's a bit much to put a full dockerfile template in an introductory part of the guide

File: docs-src/python/backgroundcheck-boilerplate-workflow-details.md (hunk 1 above)
Comment (rachfop) [reply docs-src/python/backgroundcheck-boilerplate-workflow-details.md:23]: Fixed.

File: docs-src/python/backgroundcheck-boilerplate-add-test-framework.md [hunk 20]
```diff
@@ -0,0 +1,17 @@
...
+- developer guide
+- test framework
+- python sdk
+---
+
+<!-- DO NOT EDIT THIS FILE DIRECTLY.
+THIS FILE IS GENERATED from https://github.com/temporalio/documentation-samples-python/blob/bgc/backgroundcheck_boilerplate/tests/workflow_dacx_test.py. -->
+
+Each Temporal SDK has a testing suite that can be used in conjunction with a typical language specific testing framework.
+In the Temporal Python SDK, the testing package (https://python.temporal.io/temporalio.testing.html) provides atest environment in which the Workflow and Activity code may be run for test purposes.
```
[the remark this answers is not in the document]
Comment (rachfop) [reply docs-src/python/backgroundcheck-boilerplate-add-test-framework.md:17]: Fixed.

File: docs-src/python/backgroundcheck-boilerplate-add-workflow-test-details.md [hunk 21]
```diff
@@ -0,0 +1,23 @@
...
+tags:
+- testing
+- developer guide
+- go sdk
+---
+
+<!-- DO NOT EDIT THIS FILE DIRECTLY.
+THIS FILE IS GENERATED from https://github.com/temporalio/documentation-samples-python/blob/bgc/backgroundcheck_boilerplate/tests/workflow_dacx_test.py. -->
+
+This is a unit test written in Python using the pytest library.
```
[the remark this answers is not in the document]
Comment (rachfop) [reply docs-src/python/backgroundcheck-boilerplate-add-workflow-test-details.md:15]: Removed reference to unit test.

File: docs-src/python/backgroundcheck-boilerplate-cloud-worker-details.md [hunk 22]
```diff
@@ -0,0 +1,19 @@
...
+
+<!-- DO NOT EDIT THIS FILE DIRECTLY.
+THIS FILE IS GENERATED from https://github.com/temporalio/documentation-samples-python/blob/bgc/backgroundcheck_boilerplate/cloud_worker/main_dacx.py. -->
+
+When specifying the Temporal Cloud Namespace, make sure to append the Account Id as it appears in the url of the Cloud UI.
+Consider the following Namespace url: https://cloud.temporal.io/namespaces/backgroundcheck-app.1a23b/workflows, if your Namespace is "backgroundcheck-app" and your Account Id is "1a23b", then you would specify your Namespace as "backgroundcheck-app.1a23b".
+
+The Temporal Cloud gRPC connection address includes your [Namesapce](https://python.temporal.io/temporalio.client.Client.html#namespace) and a port number: `<Namespace>.<AccountId>.tmprl.cloud:<port>`.
+For example: `https://backgroundcheck-app.1a23b.tmprl.cloud:1234`.
+There is an option to copy the grPC endpoint address from the Temporal Cloud UI.
```
[the remark this answers is not in the document]
Comment (rachfop) [reply docs-src/python/backgroundcheck-boilerplate-cloud-worker-details.md:19]: Fixed.

File: docs-src/python/backgroundcheck-boilerplate-self-hosted-worker.md (hunk 8 above)
Comment (rachfop) [reply docs-src/python/backgroundcheck-boilerplate-self-hosted-worker.md:29]: removed.

File: docs-src/python/backgroundcheck-boilerplate-self-hosted-worker.md (hunk 7 above)
Comment (rachfop) [reply docs-src/python/backgroundcheck-boilerplate-self-hosted-worker.md:27]: Removed.

File: docs-src/python/backgroundcheck-boilerplate-self-hosted-worker.md (hunk 6 above)
Comment (rachfop) [reply docs-src/python/backgroundcheck-boilerplate-self-hosted-worker.md:26]: This is used because the main is being ran.

File: docs-src/python/backgroundcheck-boilerplate-ssntrace-activity.md [hunk 23]
```diff
@@ -0,0 +1,34 @@
...
+tags:
+- go sdk
+- code sample
+- activity
+---
+
+<!-- DO NOT EDIT THIS FILE DIRECTLY.
+THIS FILE IS GENERATED from https://github.com/temporalio/documentation-samples-python/blob/bgc/backgroundcheck_boilerplate/activities/ssntraceactivity_dacx.py. -->
+
+In the Temporal Python SDK programming model, an Activity is an function.
```
[the remark this answers is not in the document]
Comment (rachfop) [reply docs-src/python/backgroundcheck-boilerplate-ssntrace-activity.md:15]: Oh nice call.

[the remark this answers is not in the document]
Comment (rachfop) [reply docs-src/python/backgroundcheck-boilerplate-ssntrace-activity.md:15]: In the Temporal Python SDK programming model, an Activity is a function and can be used as an instance method of a class

File: docs-src/python/backgroundcheck-boilerplate-ssntrace-activity.md (hunk 9 above)
Comment (rachfop) [reply docs-src/python/backgroundcheck-boilerplate-ssntrace-activity.md:16]: Fixed.

File: docs-src/python/choose-dev-cluster.md (hunk 11 above)
Comment (rachfop) [reply docs-src/python/choose-dev-cluster.md:25]: Removing smart quotes.

File: docs-src/python/choose-dev-cluster.md (hunk 13 above)
Comment (rachfop) [reply docs-src/python/choose-dev-cluster.md:58]: Great feedback.

File: docs-src/python/backgroundcheck-boilerplate-self-hosted-worker.md (hunk 6 above)
Comment (cretz) [reply docs-src/python/backgroundcheck-boilerplate-self-hosted-worker.md:26]: There is no `main` in this snippet

Comment (cretz) [reply docs-src/python/backgroundcheck-boilerplate-self-hosted-worker.md:26]: There is no code in this snippet using this import though (i.e. there is no `asyncio.run(main())` like there is elsewhere). You don't have to import `asyncio` just to defined `async def main`

File: docs-src/python/choose-dev-cluster.md (hunk 2 above)
Comment (rachfop) [reply docs-src/python/choose-dev-cluster.md:49]:
```bash
temporal server start-dev --db-filename temporal.db
```

This command automatically starts the Temporal Web UI, creates a default Namespace, and creates a persistence database.

File: docs-src/python/backgroundcheck-boilerplate-cloud-worker-details.md [hunk 24]
```diff
@@ -0,0 +1,19 @@
...
+tags:
+- worker
+- cloud certificate
+---
+
+<!-- DO NOT EDIT THIS FILE DIRECTLY.
+THIS FILE IS GENERATED from https://github.com/temporalio/documentation-samples-python/blob/bgc/backgroundcheck_boilerplate/cloud_worker/main_dacx.py. -->
+
+When specifying the Temporal Cloud Namespace, make sure to append the Account Id as it appears in the url of the Cloud UI.
+Consider the following Namespace url: https://cloud.temporal.io/namespaces/backgroundcheck-app.1a23b/workflows, if your Namespace is "backgroundcheck-app" and your Account Id is "1a23b", then you would specify your Namespace as "backgroundcheck-app.1a23b".
```
[the remark this answers is not in the document]
Comment (rachfop) [reply docs-src/python/backgroundcheck-boilerplate-cloud-worker-details.md:15]: Fixed.

File: docs-src/python/choose-dev-cluster.md (hunk 2 above)
Comment (cretz) [reply docs-src/python/choose-dev-cluster.md]: I'm not sure in Python setup is the place to document the different options of the CLI's start-dev (even if this is a common one)

File: docs-src/python/backgroundcheck-boilerplate-cloud-worker-details.md (hunk 24 above)
[the remark this answers is not in the document]
Comment (rachfop) [reply docs-src/python/backgroundcheck-boilerplate-cloud-worker-details.md]: Will surface this in a larger look across docs.

File: docs-src/python/backgroundcheck-boilerplate-workflow-details.md (hunk 10 above)
Comment (rachfop) [reply docs-src/python/backgroundcheck-boilerplate-workflow-details.md:21]: Surfaced this in a larger convo.

File: docs-src/python/install-cli.md (hunk 14 above)
Comment (rachfop) [reply docs-src/python/install-cli.md:78]: likely to stay.

File: docs-src/python/project-structure.md (hunk 15 above)
Comment (rachfop) [reply docs-src/python/project-structure.md:32]: Good point. I think this is more of a reference to get people thinking of the overall project structure and not the "go to 100% correct" idea.

File: docs-src/python/project-structure.md (hunk 17 above)
Comment (rachfop) [reply docs-src/python/project-structure.md:53]: Weird. Doesn't show up on my local machine or render view.

File: docs-src/python/self-hosted-worker-dockerfile.md
Comment (rachfop) [reply docs-src/python/self-hosted-worker-dockerfile.md:1]: Feedback surfaced to the larger project structure.

**Review by flossypurse — APPROVED** [review]
